### ISSUE TITLE
Split warnings from errors in the scheck

### ISSUE THREAD

**Issue by berezovskyi** [oslc-op/oslc-specs#41]
Here is a sample report:

> A total of 32 issues were found
> 
> Checked vocabulary http://open-services.net/ns/core# from file:/home/circleci/oslc-specs/tools/ShapeChecker/../../specs/core/vocab/core-vocab.ttl, with 0 issues
>
> ...
> 
> Checked shapes from file:/home/circleci/oslc-specs/tools/ShapeChecker/../../specs/core/shapes/CommonProperties-shape.ttl, with 1 issue
>    While examining file:/home/circleci/oslc-specs/tools/ShapeChecker/../../specs/core/shapes/CommonProperties-shape.ttl:
>       On http://open-services.net/ns/core#describes: The named property should be specified for this resource.
> 
> These classes are referenced in the given shapes, but not defined in the expected vocabulary:
>    http://xmlns.com/foaf/0.1/Person
> 
> These terms were defined in the given vocabularies, but were not referenced in the given shapes:
>    http://open-services.net/ns/core#Exactly-one
>    http://open-services.net/ns/core#ResourceValueType
>    http://open-services.net/ns/core#results
>    http://open-services.net/ns/core#default
>    http://open-services.net/ns/core#NoImpact
>    http://open-services.net/ns/core#Cardinality
>    http://open-services.net/ns/core#impactType
>    http://open-services.net/ns/core#One-or-many
>    http://open-services.net/ns/core#Reference
>    http://open-services.net/ns/core#Inline
>    http://open-services.net/ns/core#attachment
>    http://open-services.net/ns/core#archived
>    http://open-services.net/ns/core#LocalResource
>    http://open-services.net/ns/core#AnyResource
>    http://open-services.net/ns/core#Representation
>    http://open-services.net/ns/core#inverseLabel
>    http://open-services.net/ns/core#futureAction
>    http://open-services.net/ns/core#ImpactFollowsLink
>    http://open-services.net/ns/core#Zero-or-many
>    http://open-services.net/ns/core#Resource
>    http://open-services.net/ns/core#Zero-or-one
>    http://open-services.net/ns/core#Any
>    http://open-services.net/ns/core#AttachmentContainer
>    http://open-services.net/ns/core#ImpactType
>    http://open-services.net/ns/core#Either
>    http://open-services.net/ns/core#SymmetricImpact
>    http://open-services.net/ns/core#initialHeight
>    http://open-services.net/ns/core#executes
>    http://open-services.net/ns/core#ImpactOppositeLink
>    http://open-services.net/ns/core#partOfDiscussion

Do we have to count all of those as errors? Also, FOAF error is wrong:

```
> curl http://xmlns.com/foaf/0.1/ -H "Accept: application/rdf+xml" --connect-timeout 1 --max-time 5
<!-- This is the FOAF formal vocabulary description, expressed using W3C RDFS and OWL markup. foaf/spec version -->
...
<!-- FOAF classes (types) are listed first. -->

  <rdfs:Class rdf:about="http://xmlns.com/foaf/0.1/Person" rdfs:label="Person" rdfs:comment="A person." vs:term_status="stable">
...
```

**Comment by ndjc** [comment]
I think the foaf:Person error is because if you request that specific resource, you get a 303 See Other redirect to the foaf vocabulary page, and ShapeChecker does not handle that. It works better with # terms than / ones. But still a defect in ShapeChecker - worth separating out to a new issue, I think.

**Comment by ndjc** [comment]
Yes, having a distinction between warnings and errors would be useful.

**Comment by ndjc** [comment]
The issue with oslc:describes on shapes is already noted in a TODO in the source. While this property is not strictly required, it should be used almost everywhere, and so I made SC warn about its absence.  In this specific case, the CommonProperties shape is a fake shape - it does not describe any actual resource, but rather is just a placeholder to have a table of common properties, so a value of oslc:describes would be misleading. We could introduce some sort of hack, such as allowing a value of rdf:nil for an oscl:describes values that is deliberately not set to a real value.

**Comment by ndjc** [comment]
The unreferenced terms are just advisory - there are many good reasons to have vocabulary terms not used in the resource shapes. But since it might also report obsolete terms that should have been removed or marked as archaic, the report is useful. We might want to add some way of indicating which of these unreferenced terms have already been checked, and report on only new unreferenced items.

**Comment by jamsden** [comment]
CommonProperties shape could describe oslc:Resource.

**Comment by ndjc** [comment]
oslc:Resource is not a class - it is an enumeration value, an instance of the class oslc:ResourceValueType. The closest we have to a universal object class is rdfs:Resource, but since we do not rely on inferencing this doesn't add any real value over rdf:nil - and even with inferencing almost no RDF classes explicitly say they are a subclass of rdfs:Resource.

**Comment by jamsden** [comment]
I meant oslc_am:Resource which should have been in core as the root superclass of all resource classes. The enumeration literal oslc:Resource is probably an unfortunate mistake - using such a good name for such a limited purpose. But there's little that can be done about it now.

**Comment by berezovskyi** [comment]
I saw @ndjc submitting a PR but he merged it before I could review. Is it related to this issue?

**Comment by ndjc** [comment]
Yes, it was. The separate branch had been open a while, and Jim and I discussed it last Thursday, and we decided to go ahead with the merge. I meant to close this issue when I did the merge.

**Comment by ndjc** [comment]
Note that the foaf issue noted above is different - I'm not completely sure what's going on there, but as mentioned above it might be related to redirect handling.

**Comment by berezovskyi** [comment]
> Yes, it was. The separate branch had been open a while, and Jim and I discussed it last Thursday, and we decided to go ahead with the merge.

Great, thank you!

> I meant to close this issue when I did the merge.

In the future you can just add "Closes #41" to the PR comment and Github will do that for you ;)
